cpp/src/slice2cs/Gen.cpp

```diff
@@ -2372,134 +2373,273 @@ Slice::Gen::ProxyVisitor::visitOperation(const OperationPtr& operation)
-        _out << eb;
-    }
 
-    outParams = getAllOutParams(operation, "iceP_");
-    {
-        //
-        // Write the Async method implementation.
-        //
-        _out << sp;
-        _out << nl << "private " << resultTask(operation, ns, false) << " " << internalName << spar
-             << getInvocationParamsAMI(operation, ns, false, "iceP_")
-             << "bool synchronous"
-             << epar;
-        _out << sb;
-
-        string resultT = resultType(operation, ns, false);
-
-        _out << nl << "var request = " << getUnqualified("Ice.OutgoingRequestFrame", ns);
-        if(inParams.size() == 0)
+        _out << nl << resultTask(operation, ns, false) << " "
+             << asyncName << spar << getInvocationParamsAMI(operation, ns, true) << epar << " => ";
+        _out.inc();
+        if(inParams.size() == 0 && outParams.size() == 0)
         {
-            _out << ".WithEmptyParamList("
-                 << "this, \"" << operation->name() << "\", " << (isIdempotent(operation) ? "true" : "false")
-                 << ", context);";
+            _out << nl << "IceInvokeAsync(\"" << operation->name() << "\", idempotent:"
+                 << (isIdempotent(operation) ? "true" : "false") << ", "
+                 << context << ", " << progress << ", " << cancel << ");";
         }
         else
         {
-            _out << ".WithParamList("
-                 << "this, \"" << operation->name() << "\", " << (isIdempotent(operation) ? "true" : "false")
-                 << ", " << opFormatTypeToString(operation, ns) << ", context, ";
-            _out << toTuple(inParams, "iceP_") << ", " << writer << ");";
+            _out << nl << requestProperty << ".InvokeAsync(this, ";
+            if(inParams.size() > 0)
+            {
+                _out << toTuple(inParams) << ", ";
+            }
+            _out << context << ", " << progress << ", " << cancel << ");";
         }
+        _out.dec();
+    }
 
-        _out << nl << "var completed = new global::IceInternal.OperationTaskCompletionCallback<"
-             << (outParams.empty() ? "object" : resultT) << ">(progress, cancel);";
-        _out << nl << "var outAsync = new global::IceInternal.OutgoingAsyncT<" << (outParams.empty() ? "object" : resultT)
-             << ">(this, completed, request);";
+    outParams = getAllOutParams(operation, "iceP_", true);
 
-        _out << nl << "outAsync.Invoke(";
+    // Write the static outgoing request instance
+    _out << sp;
+    if(outParams.size() > 0 && inParams.size() == 0)
+    {
+        _out << nl << "private static Ice.OutgoingRequestWithEmptyParamList<" << toTupleType(outParams) << ">? "
+             << requestObject << ";";
+
+        _out << nl << "private static Ice.OutgoingRequestWithEmptyParamList<" << toTupleType(outParams)
+             << "> " << requestProperty;
+        _out << sb;
+        _out << nl << "get";
+        _out << sb;
+        _out << nl << "if (" << requestObject << " == null)";
```

Comment:
Why not use a `??=` here to initialize the value if null?
Maybe you could even reduce it to a:
``` => _iceI_xxxRequestValue ??= new OutgoingRequest...;```?

Reply:
`=> _iceI_xxxRequestValue ??= new OutgoingRequest...;`
Seems to work fine